### ISSUE TITLE
13 alpha, second invocation of handleSubmit doesn't return function

### ISSUE THREAD

**Issue by AVVS** [erikras/redux-form#986]
After initial form submit, handleSubmit(fn) doesn't return new function on rerender
Consider following snippet of code:

```js
handleRestore = (user) => {
    const { props } = this;
    return props
      .handleRestore(user.id, props.http)
      .then(data => {
        if (!data.error) {
          return this.handleSuccess(data.payload);
        }

        return props.showToast(get(data, 'payload.data.errors[0].title', data.payload));
      });
  }

  handleSuccess = (data) => {
    this.props.showToast('password was sent to your email');
    this.props.pushState('/');
    return data;
  }

  render() {
    const { handleSubmit, ...rest } = this.props;

    return (
      <PaperForm
        {...rest}
        title="Restore Password"
        handleSubmit={handleSubmit(this.handleRestore)}
        submitLabel="Request Password"
        secondaryLabel="sign in"
        to="/"
      >
        <Field name="id" component={RestorePassword.emailField} />
      </PaperForm>
    );
  }
```

**Comment by AVVS** [comment]
Basically it returns here: https://github.com/erikras/redux-form/blob/v6.0.0-alpha.13/src/reduxForm.js#L216-L218, whats missing here is check whether its a function or event, because for event it would be good to return here, even though you still need to e.preventDefault() it, but for submit function it must be (re)wrapped

**Comment by erikras** [comment]
Excellent reporting. 👍

**Comment by erikras** [comment]
@AVVS It'd be great if you could confirm this fix.

**Comment by AVVS** [comment]
@erikras can you do alpha.14 release so that I can properly test it without manually building it?

I have somewhat monkey-patched my own code to handle it now:

```js
  constructor(props, context) {
    super(props, context);
    this.state = { handleSubmit: props.handleSubmit };
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.handleSubmit) {
      return;
    }

    this.setState({ handleSubmit: nextProps.handleSubmit });
  }
```

**Comment by halayli** [comment]
I think this will also fix #966

**Comment by erikras** [comment]
I want to wait for the fallout of #961 to be implemented before 6a14. There's quite a bit of work to do.

**Comment by erikras** [comment]
Fix released in [`v6.0.0-alpha.14`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.14). Please confirm, @AVVS.
